packages/pwa-buildpack/lib/Utilities/serve.js

```diff
@@ -34,62 +29,57 @@ async function serve() {
                     ...config.section('imageOptimizing'),
                     ...config.section('imageService')
                 });
-                app.use(bestPractices());
             }
         }
     );
 
     let envPort;
     if (process.env.PORT) {
-        console.log(`PORT is set in environment: ${process.env.PORT}`);
+        prettyLogger.info(`PORT is set in environment: ${process.env.PORT}`);
         envPort = process.env.PORT;
     } else if (stagingServerSettings.port) {
-        console.log(
+        prettyLogger.info(
             `STAGING_SERVER_PORT is configured: ${stagingServerSettings.port}`
         );
         envPort = stagingServerSettings.port;
     }
 
     if (config.isProd) {
-        console.log(
+        prettyLogger.info(
             `NODE_ENV=production, will not attempt to use custom host or port`
         );
 
         if (envPort) {
             upwardServerOptions.port = envPort;
         } else {
-            console.log(`No port set. Binding to random open port`);
+            prettyLogger.warn(`No port set. Binding to random open port`);
             upwardServerOptions.port = 0;
         }
     } else {
         try {
             // don't require configureHost until you need to, since loading
             // the devcert library can have side effects.
-            const {
-                Utilities: { configureHost }
-            } = require('@magento/pwa-buildpack');
+            const configureHost = require('./configureHost');
+
             const { hostname, ports, ssl } = await configureHost(
                 Object.assign(config.section('customOrigin'), {
-                    dir: __dirname,
+                    dir: dirname,
                     interactive: false
                 })
             );
             upwardServerOptions.host = hostname;
             upwardServerOptions.https = ssl;
             upwardServerOptions.port = envPort || ports.staging || 0;
         } catch (e) {
-            console.log(
+            prettyLogger.error(
                 'Could not configure or access custom host. Using loopback...',
                 e.message
             );
         }
     }
-    console.log('Launching UPWARD server\n');
-    await createUpwardServer(upwardServerOptions);
-    console.log('\nUPWARD server running.');
-}
 
-serve().catch(e => {
-    console.error(e.stack);
-    process.exit(1);
-});
+    const { createUpwardServer } = require('@magento/upward-js');
+    prettyLogger.info('Launching UPWARD server\n');
+    await createUpwardServer(upwardServerOptions);
+    prettyLogger.success('\nUPWARD server running.');
```

Comment:
```suggestion
    const server = await createUpwardServer({
        ...upwardServerOptions,
        bindLocal: false,
        logUrl: false
    });
    prettyLogger.success('\nUPWARD server running.');
    return server;
```

The `serve.js` Node API currently returns nothing, so the API consumer has no control. It return the server tuple from `createUpwardServer` so the API consumer has a reference to the running server.